source/plugins/go/src/oms.go

```diff
@@ -1105,6 +1136,99 @@ func UpdateNumTelegrafMetricsSentTelemetry(numMetricsSent int, numSendErrors int
 	ContainerLogTelemetryMutex.Unlock()
 }
 
+func toStringMap(record map[interface{}]interface{}) map[string]interface{} {
+	tag := record["tag"].([]byte)
+	mp := make(map[string]interface{})
+	mp["tag"] = string(tag)
+	mp["messages"] = []map[string]interface{}{}
+	message := record["messages"].([]interface{})
+	for _, entry := range message {
+		newEntry := entry.(map[interface{}]interface{})
+		m := make(map[string]interface{})
+		for k, v := range newEntry {
+			switch t := v.(type) {
+			case []byte:
+				m[k.(string)] = string(t)
+			default:
+				m[k.(string)] = v
+			}
+		}
+		mp["messages"] = append(mp["messages"].([]map[string]interface{}), m)
+	}
+
+	return mp
+}
+
+func PostInputPluginRecords(inputPluginRecords []map[interface{}]interface{}) int {
+	start := time.Now()
+	var msgPackEntries []MsgPackEntry
+	Log("Info::PostInputPluginRecords starting")
+
+	for _, record := range inputPluginRecords {
+		val := toStringMap(record)
+		tag := val["tag"].(string)
+		Log("Info::PostInputPluginRecords tag: %s\n", tag)
+		messages := val["messages"].([]map[string]interface{})
+		for _, message := range messages {
+			stringMap := convertMap(message)
+			msgPackEntry := MsgPackEntry{
+				Record: stringMap,
+			}
+			msgPackEntries = append(msgPackEntries, msgPackEntry)
+		}
```

Comment:
The value of msgPackEntries is stale from the previous for loop run. Hence, if the next datatype is empty, it will still send the previous data with the new tag(whose data is empty). Declare the var msgPackEntries []MsgPackEntry inside the for loop so every time we have a local variable